src/core/components/info.jsx

```diff
@@ -100,17 +104,21 @@ export default class Info extends React.Component {
     basePath: PropTypes.string,
     externalDocs: ImPropTypes.map,
     getComponent: PropTypes.func.isRequired,
+    oas3selectors: PropTypes.func,
+    selectedServer: PropTypes.string,
   }
 
   render() {
-    let { info, url, host, basePath, getComponent, externalDocs } = this.props
+    let { info, url, host, basePath, getComponent, externalDocs, selectedServer, url: specUrl } = this.props
     let version = info.get("version")
     let description = info.get("description")
     let title = info.get("title")
-    let termsOfService = info.get("termsOfService")
+    let termsOfServiceUrl = buildUrl(info.get("termsOfService"), specUrl, {selectedServer})
```

Comment:
This was a breaking change for us. Our OAS2 documents have inline `termsOfService` rather than a URL.

Contrary to OAS 3 with OAS2 it is not required for the `termsOfService` to be a URL (as per https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#fixed-fields-1). Hence, before _building_ this URL the `termsOfService` value should first be verified to _be_ a URL.